sstables/sstables.cc

```diff
@@ -2342,7 +2342,7 @@ future<> sstable::filesystem_storage::change_state(const sstable& sst, sstring t
         co_return; // Already there
     }
 
-    sstlog.info("Moving sstable {} to {} in {}", sst.get_filename(), to, path);
+    sstlog.info("Moving sstable {} to \"{}\" in {}", sst.get_filename(), to, path);
```

Comment:
Then it's going to look like
```
Moving sstable .../data/ks/cf-4c1bb670dc3711ed96733daf102e4aab/upload/md-1-big-Data.db to "" in ".../data/ks/cf-4c1bb670dc3711ed96733daf102e4aab/"
```
which also looks weird.

Maybe

```
-    sstlog.info("Moving sstable {} to {} in {}", sst.get_filename(), to, path);
+    sstlog.info("Moving sstable {} to {} in {}", sst.get_filename(), to ? : "datadir", path);
```
?

Reply:
sstrings don't have a boolean operator, so maybe
```
sstlog.info("Moving sstable {} to {} in {}", sst.get_filename(), to.empty() ? "datadir" : to, path);
```

Also, since with object storage there is no actual move involved, how about:
```
sstlog.info("Changing state of sstable {} to {} in {}", sst.get_filename(), to.empty() ? "datadir" : to, path);
```

Note that with sstables on local storage `get_filename()` contains the sstable path, so it's repeated in this message.
What does `get_filename()` return for sstables on object storage?
Should we specialize the cases and print different log messages in either case?

Reply:
> sstrings don't have a boolean operator

:disappointed: 

> so maybe
> 
>        sstlog.info("Moving sstable {} to {} in {}", sst.get_filename(), to.empty() ? "datadir" : to, path);

Yes, I meant this.

> Also, since with object storage there is no actual move involved, how about:
> 
>       sstlog.info("Changing state of sstable {} to {} in {}", sst.get_filename(), to.empty() ? "datadir" : to, path);

Then it's worth moving the message, because currently it sits in `sstable::filesystem_storage::change_state` which's not called for S3

> What does get_filename() return for sstables on object storage?

It will return `location + / + component_name` where `location` is whatever passed into sstable constructor. Currentl it's full path, but once #12707 is done it will be just the trailing part of it, like `keyspace_name + / + column_familiy_name + _ + column_family_uuid`

> Should we specialize the cases and print different log messages in either case?

Depending on the verbosity level we need for debugging. I would say that putting one message in `sstable::change_state` would be enough.

Reply:
@xemul so at filesystem_storage layer what's wrong with simply printing?
```
sstlog.info("Moving sstable {} to {}", sst.get_filename(), path / to);
```

Reply:
I'm looking at the code and see that `path` already contains `to`, so it can be even as simple as

```
sstlog.info("Moving sstable {} to {}", sst.get_filename(), path);
```

Reply:
Simplified as proposed in v2